CMake error while building app

I am running the following command to build my basic app (generated by `flutter-elinux create .`) : 

```
flutter-elinux build elinux --target-arch=arm64 --target-sysroot=/opt/arm64-sysroot \
    --system-include-directories=/usr/aarch64-linux-gnu/include/c++/9/aarch64-linux-gnu \
    --debug
```

And i have the following output : 

```
tom@tom-VirtualBox:~/StudioProjects/test$ flutter-elinux build elinux --target-arch=arm64 --target-sysroot=/opt/arm64-sysroot     --system-include-directories=/usr/aarch64-linux-gnu/include/c++/9/aarch64-linux-gnu     --debug
Flutter 3.0.5 • channel unknown • unknown source
Framework • revision f1875d570e (4 weeks ago) • 2022-07-13 11:24:16 -0700
Engine • revision e85ea0e79c
Tools • Dart 2.17.6 • DevTools 2.12.2

💪 Building with sound null safety 💪

Building an eLinux application with wayland backend in debug mode for arm64 target...      1,365ms
Failed to cmake:
-- The CXX compiler identification is Clang 14.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ - broken
-- Configuring incomplete, errors occurred!
See also "/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeOutput.log".
See also "/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeError.log".

CMake Error at /usr/share/cmake-3.22/Modules/CMakeTestCXXCompiler.cmake:62 (message):
  The C++ compiler

    "/usr/bin/clang++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp

    Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_84996/fast && /usr/bin/gmake  -f
    CMakeFiles/cmTC_84996.dir/build.make CMakeFiles/cmTC_84996.dir/build
    gmake[1]: Entering directory '/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp'
    Building CXX object CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o
    /usr/bin/clang++ --target=aarch64-linux-gnu --sysroot=/opt/arm64-sysroot    -MD -MT
    CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o -MF CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o.d -o
    CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o -c
    /home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp/testCXXCompiler.cxx
    Linking CXX executable cmTC_84996
    /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_84996.dir/link.txt --verbose=1
    /usr/bin/clang++ --target=aarch64-linux-gnu --sysroot=/opt/arm64-sysroot
    CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o -o cmTC_84996
    /usr/bin/ld: unrecognised emulation mode: aarch64linux
    Supported emulations: elf_x86_64 elf32_x86_64 elf_i386 elf_iamcu elf_l1om elf_k1om i386pep i386pe
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    gmake[1]: *** [CMakeFiles/cmTC_84996.dir/build.make:100: cmTC_84996] Error 1
    gmake[1]: Leaving directory '/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp'
    gmake: *** [Makefile:127: cmTC_84996/fast] Error 2


  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:2 (project)
```

And i have `cat` the file `/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeError.log` : 

```
tom@tom-VirtualBox:~/StudioProjects/test$ cat /home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeError.log
Detecting CXX compiler ABI info failed to compile with the following output:
Change Dir: /home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_0a0fb/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_0a0fb.dir/build.make CMakeFiles/cmTC_0a0fb.dir/build
gmake[1]: Entering directory '/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o
/usr/bin/clang++ --target=aarch64-linux-gnu --sysroot=/opt/arm64-sysroot   -v -MD -MT CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o -MF CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o.d -o CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o -c /usr/share/cmake-3.22/Modules/CMakeCXXCompilerABI.cpp
Ubuntu clang version 14.0.0-1ubuntu1
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
 (in-process)
 "/usr/lib/llvm-14/bin/clang" -cc1 -triple aarch64-unknown-linux-gnu -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name CMakeCXXCompilerABI.cpp -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=non-leaf -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu generic -target-feature +neon -target-feature +v8a -target-abi aapcs -fallow-half-arguments-and-returns -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=gdb -v -fcoverage-compilation-dir=/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp -resource-dir /usr/lib/llvm-14/lib/clang/14.0.0 -dependency-file CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o.d -MT CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o -sys-header-deps -isysroot /opt/arm64-sysroot -internal-isystem /usr/lib/llvm-14/lib/clang/14.0.0/include -internal-isystem /opt/arm64-sysroot/usr/local/include -internal-externc-isystem /opt/arm64-sysroot/include -internal-externc-isystem /opt/arm64-sysroot/usr/include -fdeprecated-macro -fdebug-compilation-dir=/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp -ferror-limit 19 -fno-signed-char -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o -x c++ /usr/share/cmake-3.22/Modules/CMakeCXXCompilerABI.cpp
clang -cc1 version 14.0.0 based upon LLVM 14.0.0 default target x86_64-pc-linux-gnu
ignoring nonexistent directory "/opt/arm64-sysroot/usr/local/include"
ignoring nonexistent directory "/opt/arm64-sysroot/include"
ignoring nonexistent directory "/opt/arm64-sysroot/usr/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/llvm-14/lib/clang/14.0.0/include
End of search list.
Linking CXX executable cmTC_0a0fb
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_0a0fb.dir/link.txt --verbose=1
/usr/bin/clang++ --target=aarch64-linux-gnu --sysroot=/opt/arm64-sysroot  -v CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o -o cmTC_0a0fb 
Ubuntu clang version 14.0.0-1ubuntu1
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
 "/usr/bin/ld" --sysroot=/opt/arm64-sysroot -pie -EL -z relro --hash-style=gnu --build-id --eh-frame-hdr -m aarch64linux -dynamic-linker /lib/ld-linux-aarch64.so.1 -o cmTC_0a0fb Scrt1.o crti.o crtbeginS.o CMakeFiles/cmTC_0a0fb.dir/CMakeCXXCompilerABI.cpp.o -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc crtendS.o crtn.o
/usr/bin/ld: unrecognised emulation mode: aarch64linux
Supported emulations: elf_x86_64 elf32_x86_64 elf_i386 elf_iamcu elf_l1om elf_k1om i386pep i386pe
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[1]: *** [CMakeFiles/cmTC_0a0fb.dir/build.make:100: cmTC_0a0fb] Error 1
gmake[1]: Leaving directory '/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:127: cmTC_0a0fb/fast] Error 2


Determining if the CXX compiler works failed with the following output:
Change Dir: /home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_84996/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_84996.dir/build.make CMakeFiles/cmTC_84996.dir/build
gmake[1]: Entering directory '/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o
/usr/bin/clang++ --target=aarch64-linux-gnu --sysroot=/opt/arm64-sysroot    -MD -MT CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o -MF CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o.d -o CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o -c /home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp/testCXXCompiler.cxx
Linking CXX executable cmTC_84996
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_84996.dir/link.txt --verbose=1
/usr/bin/clang++ --target=aarch64-linux-gnu --sysroot=/opt/arm64-sysroot CMakeFiles/cmTC_84996.dir/testCXXCompiler.cxx.o -o cmTC_84996 
/usr/bin/ld: unrecognised emulation mode: aarch64linux
Supported emulations: elf_x86_64 elf32_x86_64 elf_i386 elf_iamcu elf_l1om elf_k1om i386pep i386pe
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[1]: *** [CMakeFiles/cmTC_84996.dir/build.make:100: cmTC_84996] Error 1
gmake[1]: Leaving directory '/home/tom/StudioProjects/test/build/elinux/arm64/debug/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:127: cmTC_84996/fast] Error 2

```

I have installed all the mandatory dependencies : 

```
$ sudo apt install curl unzip git clang cmake pkg-config
$ sudo apt install libegl1-mesa libgles2-mesa libxkbcommon-dev
```

Thanks in advance for your help

I have got the following error using the docker solution : 

```
tom@tom-VirtualBox:~/test_app$ flutter-elinux build elinux --target-arch=arm64 --target-sysroot=/home/tom/ubuntu18-arm64-sysroot --system-include-directories=/usr/aarch64-linux-gnu/include/c++/9/aarch64-linux-gnu --debug
Flutter 3.0.5 • channel unknown • unknown source
Framework • revision f1875d570e (4 weeks ago) • 2022-07-13 11:24:16 -0700
Engine • revision e85ea0e79c
Tools • Dart 2.17.6 • DevTools 2.12.2

💪 Building with sound null safety 💪

Building an eLinux application with wayland backend in debug mode for arm64 target...      1,658ms
Failed to cmake:
-- The CXX compiler identification is unknown
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /home/tom/ubuntu18-arm64-sysroot/usr/bin/clang++
-- Check for working CXX compiler: /home/tom/ubuntu18-arm64-sysroot/usr/bin/clang++ - broken
-- Configuring incomplete, errors occurred!
See also "/home/tom/test_app/build/elinux/arm64/debug/CMakeFiles/CMakeOutput.log".
See also "/home/tom/test_app/build/elinux/arm64/debug/CMakeFiles/CMakeError.log".

CMake Error at /usr/share/cmake-3.22/Modules/CMakeTestCXXCompiler.cmake:62 (message):
  The C++ compiler

    "/home/tom/ubuntu18-arm64-sysroot/usr/bin/clang++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/tom/test_app/build/elinux/arm64/debug/CMakeFiles/CMakeTmp

    Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_848af/fast && /usr/bin/gmake  -f
    CMakeFiles/cmTC_848af.dir/build.make CMakeFiles/cmTC_848af.dir/build
    gmake[1]: Entering directory '/home/tom/test_app/build/elinux/arm64/debug/CMakeFiles/CMakeTmp'
    Building CXX object CMakeFiles/cmTC_848af.dir/testCXXCompiler.cxx.o
    /home/tom/ubuntu18-arm64-sysroot/usr/bin/clang++    -o CMakeFiles/cmTC_848af.dir/testCXXCompiler.cxx.o -c
    /home/tom/test_app/build/elinux/arm64/debug/CMakeFiles/CMakeTmp/testCXXCompiler.cxx
    aarch64-binfmt-P: Could not open '/lib/ld-linux-aarch64.so.1': No such file or directory
    gmake[1]: *** [CMakeFiles/cmTC_848af.dir/build.make:78: CMakeFiles/cmTC_848af.dir/testCXXCompiler.cxx.o] Error 255
    gmake[1]: Leaving directory '/home/tom/test_app/build/elinux/arm64/debug/CMakeFiles/CMakeTmp'
    gmake: *** [Makefile:127: cmTC_848af/fast] Error 2





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:2 (project)
```

--- a/lib/elinux_build_target.dart
+++ b/lib/elinux_build_target.dart
@@ -375,11 +375,10 @@ class NativeBundle {
         if (targetCompilerFlags != null) '-DCMAKE_C_FLAGS=$targetCompilerFlags',
         if (targetCompilerFlags != null)
           '-DCMAKE_CXX_FLAGS=$targetCompilerFlags',
-        '-DCMAKE_C_COMPILER=clang',
-        '-DCMAKE_CXX_COMPILER=clang++',
         eLinuxDir.path,
       ],
       workingDirectory: outputDir.path,
+      environment: <String, String>{'CC': 'clang', 'CXX': 'clang++'},
     );
     if (result.exitCode != 0) {
       throwToolExit('Failed to cmake:\n$result');

